tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -1231,6 +1231,70 @@ def get_kurtosis(y):
     return zip(index, res)
 
 
+@set_property("fctype", "combiner")
+@set_property("input", "pd.Series")
+@set_property("index_type", pd.DatetimeIndex)
+def energy_content_frequency_brackets(
+    signal: pd.Series, params: dict = {"number_of_bins": 10, "with_normalization": True}
+) -> list[tuple[str, float]]:
+    """
+
+    Calculates the FFT of the signal and groups the content of the FFT into n bins, \
+        to get an estimate of energy content per frequency bucket
+
+    Args:
+        signal (pd.Series): The distribution for which the frequency energy buckets \
+            should be calculated
+        params (_type_, optional): Parameters used for calculation of the frequency \
+            energy buckets. Parameters to be set are number_of_bins for the amount \
+                of bins, and if the values should be normalized (divided by the size \
+                    of the frequency bucket) or not.\
+            Defaults to {"number_of_bins": 10, "with_normalization": True}.
+
+    Returns:
+        list[tuple[str, float]]: _description_
+    """
+    # pylint:disable=dangerous-default-value
+
+    time_vec: np.ndarray = signal.index.to_numpy()
+    number_of_bins: int = 10
+    if "number_of_bins" in params.keys():
+        number_of_bins: int = (
+            params["number_of_bins"]
+            if (
+                params["number_of_bins"] <= len(time_vec)
+                and params["number_of_bins"] >= 0
+            )
+            else len(time_vec) - 1
+        )
```

Comment:
You do not need to use a default value for `number_of_bins` here. You can always assume that the key will be set. The reason is that we define the settings ourself :)
This allows you to simplify the code to (basically)
```
number_of_bins = min(params["number_of_bins"], len(time_vec) - 1)
```

Reply:
I think if a user sets `number_of_bins` to 0 or below this should be a bug and you can just raise an exception here.

Reply:
I chose to raise a warning to inform the user, and to continue with a default value of 1.